types/slate/index.d.ts

```diff
@@ -1100,81 +1100,355 @@ export class Editor {
+    insertTextByKey(
+        key: string,
+        offset: number,
+        text: string,
+        marks?: Immutable.Set<Mark> | Mark[]
+    ): Editor;
+    insertTextByPath(
+        path: Path,
+        offset: number,
+        text: string,
+        marks?: Immutable.Set<Mark> | Mark[]
+    ): Editor;
+    mergeNodeByKey(key: string): Editor;
+    mergeNodeByPath(path: Path): Editor;
+    moveNodeByKey(key: string, newKey: string, newIndex: number): Editor;
+    moveNodeByPath(path: Path, newPath: Path, newIndex: number): Editor;
+    removeMarkByKey(
+        key: string,
+        offset: number,
+        length: number,
+        mark: Mark | Mark | string
+    ): Editor;
+    removeMarkByPath(
+        path: Path,
+        offset: number,
+        length: number,
+        mark: MarkProperties | Mark | string
+    ): Editor;
+    removeNodeByKey(key: string): Editor;
+    removeNodeByPath(path: Path): Editor;
+    replaceNodeByKey(key: string, node: Node): Editor;
+    replaceNodeByPath(path: Path, newNode: Node): Editor;
+    removeTextByKey(key: string, offset: number, length: number): Editor;
+    removeTextByPath(path: Path, offset: number, length: number): Editor;
+    setMarkByKey(
+        key: string,
+        offset: number,
+        length: number,
+        mark: Mark,
+        properties: MarkProperties
+    ): Editor;
+    setMarksByPath(
+        path: Path,
+        offset: number,
+        length: number,
+        mark: Mark,
+        properties: MarkProperties
+    ): Editor;
+    setNodeByKey(key: string, properties: BlockProperties | InlineProperties | string): Editor;
+    setNodeByPath(path: Path, properties: NodeProperties | InlineProperties | string): Editor;
+    splitNodeByKey(key: string, offset: number): Editor;
+    splitNodeByPath(path: Path, position: number): Editor;
+    unwrapInlineByKey(key: string, properties: InlineProperties | string): Editor;
+    unwrapInlineByPath(path: Path, properties: InlineProperties | string): Editor;
+    unwrapBlockByKey(key: string, properties: BlockProperties | string): Editor;
+    unwrapBlockByPath(path: Path, properties: BlockProperties | string): Editor;
+    unwrapNodeByKey(key: string): Editor;
+    unwrapNodeByPath(path: Path): Editor;
+    wrapInlineByKey(key: string, properties: InlineProperties | string): Editor;
+    wrapInlineByPath(path: Path, properties: InlineProperties | string): Editor;
+    wrapBlockByKey(key: string, properties: BlockProperties | string): Editor;
+    wrapBlockByPath(path: Path, block: Block | string): Editor;
+    wrapNodeByKey(key: string, parent: Node): Editor;
+    wrapNodeByPath(path: Path, parent: Node): Editor;
+    normalize(): Editor;
+    withoutNormalizing(fn: () => void): Editor;
+    withoutSaving(fn: () => void): Editor;
+    withoutMerging(fn: () => void): Editor;
+    redo(): Editor;
+    undo(): Editor;
+    snapshotSelection(): Editor;
+}
+
+export interface Commandable {
```

Comment:
anything against this being `Controller` instead? Since it also holds queries (or should: I haven't looked at it yet). Really what I think this should be is a representations of the core plugins

Reply:
Ah yeah I think `Controller` is probably a better shout, I also need to move over the additional query commands to that interface, will do that

Reply:
I've push some changes to this effect